src/robotFactory.js

```diff
@@ -1,10 +1,66 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
 
-class DeliveryDrone {}
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+  getInfo() {
+    return (
+      `Robot: ${this.name}, `
+      + `Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`
+    );
+  }
+}
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+    this.coords.z = coords.z || 0;
+  }
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion, maxLoadWeight, currentLoad) {
+    super(name, weight, coords, chipVersion);
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad || null;
+  }
+  hookLoad(cargo) {
+    if (this.currentLoad === null && cargo.weight <= this.maxLoadWeight) {
```

Comment:
Split strings with more than 80 characters,
Chek all your code for this.
Also my suggestion to change this.currentLoad === null, its would be more compact  
```suggestion
    if (
      !this.currentLoad
      && cargo.weight <= this.maxLoadWeight
    ) {
```